### PR TITLE
YLP-458 Team modal edit / create

### PR DESCRIPTION
**Pull request by yvon-dblg**

### CODE REVIEW

File: packages/yourloops/pages/hcp/team-card.tsx [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+    },
+    firstRow: {
+      display: "flex",
+      flexDirection: "row",
+      marginBottom: theme.spacing(2), // eslint-disable-line no-magic-numbers
+    },
+    secondRow: {
+      display: "flex",
+      flexDirection: "row",
+      justifyContent: "space-between",
```
Comment (franck-fourel) [inline packages/yourloops/pages/hcp/team-card.tsx:77]:
```suggestion
      justifyContent: "left",
      columnGap: "10px",
```
I know I'm not supposed to play with css ;-(

File: packages/yourloops/pages/hcp/team-edit-modal.tsx [hunk 2]
```diff
@@ -0,0 +1,240 @@
...
+  const [ addrLine1, setAddrLine1 ] = React.useState(team.address?.line1 ?? "");
+  const [ addrLine2, setAddrLine2 ] = React.useState(team.address?.line2 ?? "");
+  const [ addrZipCode, setAddrZipCode ] = React.useState(team.address?.zip ?? "");
+  const [ addrCity, setAddrCity ] = React.useState(team.address?.city ?? "");
+  const [ addrCountry, setAddrCountry ] = React.useState(auth.user?.settings?.country ?? "FR");
+  const [ formIsIncomplete, setFormIsIncomplete ] = React.useState(true);
+
+  const ariaModal = action === "create" ? t("aria-modal-team-add") : t("aria-modal-team-edit");
+  const modalTitle = action === "create" ? t("modal-team-add-title") : t("modal-team-edit-title");
+  const modalButtonValidate = action === "create" ? t("modal-team-button-create") : t("modal-team-button-edit");
```
Comment (franck-fourel) [inline packages/yourloops/pages/hcp/team-edit-modal.tsx:119]: for sure we are saving lines ;-) 
but I really prefer it written that way. 
```suggestion
  let ariaModal = t("aria-modal-team-add");
  let modalTitle = t("modal-team-add-title");
  let modalButtonValidate = t("modal-team-button-create");
  if (action !== "create") {
    ariaModal = t("aria-modal-team-edit");
    modalTitle = t("modal-team-edit-title");
    modalButtonValidate = t("modal-team-button-edit");
  }
```

File: packages/yourloops/pages/hcp/team-edit-modal.tsx [hunk 3]
```diff
@@ -0,0 +1,240 @@
...
+  const isFormIsIncomplete = (): boolean => {
+    let valid = teamName.length > 1;
+    valid = valid && teamPhone.length > 1;
+    valid = valid && addrLine1.length > 1;
+    valid = valid && addrZipCode.length > 1;
+    valid = valid && addrCity.length > 1;
+    valid = valid && addrCountry.length > 1;
+    valid = valid && (teamEmail.length < 1 || REGEX_EMAIL.test(teamEmail));
+    return !valid;
+  };
```
Comment (franck-fourel) [inline packages/yourloops/pages/hcp/team-edit-modal.tsx:144]: I would at least manage  a max length for the team Name.
But I guess it's something we may have to adjust in next versions with specific conditions and warning messages. But we can keep it as it is.

File: packages/yourloops/pages/hcp/team-edit-modal.tsx [hunk 4]
```diff
@@ -0,0 +1,240 @@
...
+
+  return (
+    <Modal id="team-modal-edit" aria-labelledby={ariaModal} className={classes.modal} open={modalOpened} BackdropComponent={Backdrop} BackdropProps={{ timeout: modalBackdropTimeout }}>
+      <Fade in={modalOpened}>
+        <div className={classes.divMain}>
+          <h2 id="team-modal-edit-title">{modalTitle}</h2>
+          <form noValidate autoComplete="off" className={classes.form}>
+            <TextField id="team-modal-edit-field-name" className={classes.formChild} variant="outlined" onChange={handleChangeTeamName} name="name" value={teamName} label={t("team-modal-edit-placeholder-name")} required={true} aria-required="true" />
+
+            <TextField id="team-modal-edit-field-line1" className={classes.formChild} variant="outlined" onChange={handleChangeAddrLine1} name="addr-line1" value={addrLine1} label={t("team-modal-edit-placeholder-addr-line1")} required={true} aria-required="true" />
```
Comment (franck-fourel) [inline packages/yourloops/pages/hcp/team-edit-modal.tsx:211]: Would that make sense to make it more readable that way ?

```suggestion
            <TextField 
              id="team-modal-edit-field-line1" 
              className={classes.formChild} 
              variant="outlined" 
              onChange={handleChangeAddrLine1} 
              name="addr-line1" 
              value={addrLine1} 
              label={t("team-modal-edit-placeholder-addr-line1")} 
              required={true} 
              aria-required="true" />
```
Same for the other similar fields.
Just for readibility ;-)

**Review by franck-fourel — COMMENTED** [review]
couple of minor comments.
Let me know if you want to mange them in this PR or in a coming PR

File: packages/yourloops/pages/hcp/team-card.tsx (hunk 1 above)
Comment (yvon-dblg) [reply packages/yourloops/pages/hcp/team-card.tsx]: Yes, I know this case.
But It need a little bit more tunning. I was planning fixing it in the last PR of this screen.

File: packages/yourloops/pages/hcp/team-edit-modal.tsx (hunk 3 above)
Comment (yvon-dblg) [reply packages/yourloops/pages/hcp/team-edit-modal.tsx:144]: Good idea.
Same for all the fields

File: packages/yourloops/pages/hcp/team-edit-modal.tsx (hunk 4 above)
Comment (yvon-dblg) [reply packages/yourloops/pages/hcp/team-edit-modal.tsx:211]: Why not.

**Pull request merged**